core/src/main/scala/org/apache/spark/ui/jobs/AllStagesPage.scala

```diff
@@ -143,76 +72,105 @@ private[ui] class AllStagesPage(parent: StagesTab) extends WebUIPage("") {
-        <span id="skipped" class="collapse-aggregated-allSkippedStages collapse-table"
-              onClick="collapseTable('collapse-aggregated-allSkippedStages',
-            'aggregated-allSkippedStages')">
-          <h4>
-            <span class="collapse-table-arrow arrow-open"></span>
-            <a>Skipped Stages ({skippedStages.size})</a>
-          </h4>
-        </span> ++
-        <div class="aggregated-allSkippedStages collapsible-table">
-          {skippedStagesTable.toNodeSeq}
-        </div>
+  }
+
+  private def tableHeaderID(status: StageStatus): String = status match {
+    case StageStatus.ACTIVE => "active"
+    case StageStatus.COMPLETE => "completed"
+    case StageStatus.FAILED => "failed"
+    case StageStatus.PENDING => "pending"
+    case StageStatus.SKIPPED => "skipped"
+  }
+
+  private def stageTag(status: StageStatus): String = status match {
+    case StageStatus.ACTIVE => "activeStage"
+    case StageStatus.COMPLETE => "completedStage"
+    case StageStatus.FAILED => "failedStage"
+    case StageStatus.PENDING => "pendingStage"
+    case StageStatus.SKIPPED => "skippedStage"
+  }
+
+  private def headerDescription(status: StageStatus): String = status match {
+    case StageStatus.ACTIVE => "Active"
+    case StageStatus.COMPLETE => "Completed"
+    case StageStatus.FAILED => "Failed"
+    case StageStatus.PENDING => "Pending"
+    case StageStatus.SKIPPED => "Skipped"
+  }
+
+  private def classSuffix(status: StageStatus): String = status match {
+    case StageStatus.ACTIVE => "ActiveStages"
+    case StageStatus.COMPLETE => "CompletedStages"
+    case StageStatus.FAILED => "FailedStages"
+    case StageStatus.PENDING => "PendingStages"
+    case StageStatus.SKIPPED => "SkippedStages"
+  }
+
+  private def summaryContent(status: StageStatus, size: Int): String = {
+    if (status == StageStatus.COMPLETE
+        && appSummary.numCompletedStages != size) {
+      s"${appSummary.numCompletedStages}, only showing $size"
+    } else {
+      s"$size"
     }
-    if (shouldShowFailedStages) {
-      content ++=
-        <span id ="failed" class="collapse-aggregated-allFailedStages collapse-table"
-            onClick="collapseTable('collapse-aggregated-allFailedStages',
-            'aggregated-allFailedStages')">
-          <h4>
-            <span class="collapse-table-arrow arrow-open"></span>
-            <a>Failed Stages ({numFailedStages})</a>
-          </h4>
-        </span> ++
-        <div class="aggregated-allFailedStages collapsible-table">
-          {failedStagesTable.toNodeSeq}
-        </div>
+  }
+
+  private def summary(status: StageStatus, size: Int): Elem = {
+    val summary =
+      <li>
+        <a href={s"#${tableHeaderID(status)}"}>
+          <strong>{headerDescription(status)} Stages:</strong>
+        </a>
+        {summaryContent(status, size)}
+      </li>
+
+    if (status == StageStatus.COMPLETE) {
+      summary % Attribute(None, "id", Text("completed-summary"), Null)
```

Comment:
In the previous code this was also the case for `SKIPPED`, are you changing that intentionally?

Reply:
yes, I realized it while doing the refactor. It was a copy-and-paste mistake, sorry.